test/module/vendor/rocksdb_test.cpp

```diff
@@ -0,0 +1,269 @@
+  std::string const value5_ = "vaLUe5";
+};
+
+TEST_F(RocksDBTest, SimpleOperation) {
+  ASSERT_TRUE(readDb(key1_) == value1_);
+  ASSERT_TRUE(readDb(key2_) == value2_);
+  ASSERT_TRUE(readDb(key3_) == value3_);
+  ASSERT_TRUE(readDb(key4_) == value4_);
+  ASSERT_TRUE(readDb(key5_) == value5_);
+}
+
+TEST_F(RocksDBTest, SimpleDelete) {
+  RocksDbCommon common(tx_context_);
+  common.del(key3_);
+
+  auto status = common.get(key3_);
+  ASSERT_TRUE(status.IsNotFound());
+}
+
+TEST_F(RocksDBTest, SimpleSeek) {
+  RocksDbCommon common(tx_context_);
+  auto it = common.seek("key");
+  ASSERT_TRUE(it->status().ok());
+
+  ASSERT_TRUE(it->key().ToStringView() == key4_);
+  ASSERT_TRUE(it->value().ToStringView() == value4_);
+
+  it = common.seek("ke1");
+  ASSERT_TRUE(it->status().ok());
+
+  ASSERT_TRUE(it->key().ToStringView() == key3_);
+  ASSERT_TRUE(it->value().ToStringView() == value3_);
+}
+
+TEST_F(RocksDBTest, SimpleEnumerateKeys) {
+  RocksDbCommon common(tx_context_);
+  int counter = 0;
+  common.enumerate(
+      [&](auto const &it, auto key_size) mutable {
+        ++counter;
+        if (it->key().ToStringView() != key1_
+            && it->key().ToStringView() != key2_)
+          throw;
+        return true;
+      },
+      "keY");
+  ASSERT_EQ(counter, 2);
+}
+
+TEST_F(RocksDBTest, SimpleEnumerateKeys2) {
+  RocksDbCommon common(tx_context_);
+  int counter = 0;
+  common.enumerate(
+      [&](auto const &it, auto key_size) mutable {
+        ++counter;
+        if (it->key().ToStringView() != key4_)
+          throw;
+        return true;
+      },
+      "key");
+  ASSERT_EQ(counter, 1);
+}
+
+TEST_F(RocksDBTest, SimpleEnumerateKeys3) {
+  RocksDbCommon common(tx_context_);
+  common.enumerate(
+      [&](auto const &it, auto key_size) mutable {
+        throw;
+        return false;
+      },
+      "keyT");
+  common.enumerate(
+      [&](auto const &it, auto key_size) mutable {
+        throw;
+        return false;
+      },
+      "ko");
+}
+
+TEST_F(RocksDBTest, SimpleRewrite) {
+  insertDb(key3_, value1_);
+  ASSERT_TRUE(readDb(key3_) == value1_);
+}
+
+TEST_F(RocksDBTest, NumberRewrite) {
+  {
+    RocksDbCommon common(tx_context_);
+    common.encode(55ull);
+    common.put("{}", "123");
+  }
+  uint64_t value;
+  {
+    RocksDbCommon common(tx_context_);
+    common.get("{}", "123");
+    common.decode(value);
+  }
+  ASSERT_TRUE(value == 55ull);
+}
+
+TEST_F(RocksDBTest, Quorum) {
+  RocksDbCommon common(tx_context_);
+  forQuorum<kDbOperation::kCheck, kDbEntry::kMustNotExist>(
+      common,
+      "dom",
+      "acc",
+      [](auto /*account*/, auto /*domain*/, auto /*opt_quorum*/) {});
+
+  common.encode(5);
+  forQuorum<kDbOperation::kPut>(
+      common,
+      "dom",
+      "acc",
+      [](auto /*account*/, auto /*domain*/, auto /*opt_quorum*/) {});
+
+  forQuorum<kDbOperation::kGet, kDbEntry::kMustExist>(
+      common, "dom", "acc", [](auto account, auto domain, auto opt_quorum) {
```

Comment:
Looks like order of domain and accout are invalid in one of places.

Reply:
Hmmm...I didn't think about perception. My point was in call position to make order in priority like in db structure and in response part in order we read this params in documentation

Reply:
I think it is important to be consistent!